### PR TITLE
Import Cloud upgrade docs

### PR DESCRIPTION
**Pull request by kilfoyle**
This imports the most recent Cloud upgrade docs into the Installation and Upgrade Guide.
This is very much a work in progress...

Closes: https://github.com/elastic/cloud/issues/92125
Rel: https://github.com/elastic/cloud/issues/92410
Rel: https://github.com/elastic/dev/issues/1808

### CODE REVIEW

**Comment by kilfoyle** [comment]
@elasticmachine update branch

**Comment by kilfoyle** [comment]
@elasticmachine update branch

**Review by lcawl — APPROVED** [review]
Functionality LGTM, but content review by @debadair is likely good idea too

**Comment by kilfoyle** [comment]
Just for quick reference: [Preview link](https://stack-docs_1876.docs-preview.app.elstc.co/guide/en/elastic-stack/master/upgrading-elastic-stack.html)

**Comment by kilfoyle** [comment]
@elasticmachine update branch

File: docs/en/install-upgrade/upgrading-stack-cloud.asciidoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
++
+If your cluster does not include a Kibana instance yet, you can {cloud}/ec-enable-kibana.html[enable Kibana first], then use the Upgrade Assistant. After you resolve the deprecation issues, try the upgrade again.
+
+Snapshot behavior during upgrades::
+To keep your data safe, a snapshot is taken automatically before any changes are made to your cluster. After a major version upgrade is complete, and the first successful snapshot with the new major version is available, all snapshots taken with the previous version of Elasticsearch are removed.
++
+For example, before upgrading from Elasticsearch version 7.x to version 8.x, a snapshot is taken. After the upgrade to version 8.x is complete, and a snapshot is available, all snapshots taken with version 8.x are removed.
```
Comment (DanRoscigno) [inline docs/en/install-upgrade/upgrading-stack-cloud.asciidoc:45]:
```suggestion
For example, before upgrading from Elasticsearch version 7.x to version 8.x, a snapshot is taken. After the upgrade to version 8.x is complete, and an 8.x snapshot is available, all snapshots taken with version 7.x are removed.
```

File: docs/en/install-upgrade/upgrading-stack-cloud.asciidoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+* In {cloud} and {ece}, for Elasticsearch 7.8 and later the Index Lifecycle Managment (ILM) API is always enabled, even if the cluster is still using the deprecated index curation. If index curation and ILM are both configured to process the same indices the outcome can be unpredictable. As other stack products default to using ILM when it is available in Elasticsearch, we strongly encourage you to {cloud}/ec-migrate-index-management.html[Migrate to index lifecycle management]. ILM also has a superior feature set.
+
+Update the security realm settings::
+During the upgrade process, you are prompted to update the security realm settings when your `user_settings` include a `xpack.security.authc.realms` value. 
++
+There are two settings that prompt you to update:
++
+* If the security realm settings are located in `user_settings_override`, contact support to help you upgrade.
++
+* If the security realm settings are located in `user_settings`, you'll receive a message that allows you modify these settings.
```
Comment (DanRoscigno) [inline docs/en/install-upgrade/upgrading-stack-cloud.asciidoc:71]:
```suggestion
* If the security realm settings are located in `user_settings`, you'll receive a message that allows you to modify these settings. 
```

**Comment by kilfoyle** [comment]
@debadair  Would you mind taking a look at the [preview](https://stack-docs_1876.docs-preview.app.elstc.co/guide/en/elastic-stack/master/upgrading-elastic-stack.html) for the upgrade stack page and letting me know if this is okay? I think this is what we discussed in our meeting about the upgrade docs, at least with regard to porting the Cloud upgrade docs into the stack, but my memory is a bit hazy. :-)

**Comment by kilfoyle** [comment]
After a terrific discussion with @debadair this is now merged.

**Pull request merged**